app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: Setting.User.name.maximum}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
giải thích dòng này?

Reply:
/ ..... / : bắt đầu và kết thúc một chuỗi regex
\A: bắt đầu một chuỗi
\w : nó là dạng ký tự thường ( chữ, số, _,)
+ : xuất hiện nhiều hơn 1
\- : ký tự phải là '-'
. :  bất kỳ một ký tự nào
i: không phân in hoa hay in thường
\d : là số
\z : kết thuc một chuỗi
\A[\w+\-.]+ :bắt đầu ký tự thường, hoặc +,- , '.'.số lần xuất hiện có thể nhiều hơn 1
@[a-z\d\-.]+: phải có ký tự @ ,sau đó ít nhất là 1 ký tự từ a-z hoặc 1 số hoặc ký tự - hoặc ký tự '.'
\.[a-z]+ : phải có ký tự . ,sau đó ít nhất một ký tự từ a-z